src/hotspot/share/runtime/os.cpp

```diff
@@ -621,36 +621,68 @@ static void break_if_ptr_caught(void* ptr) {
 }
 #endif // ASSERT
 
-void* os::malloc(size_t size, MemTag mem_tag) {
-  return os::malloc(size, mem_tag, CALLER_PC);
-}
-
-void* os::malloc(size_t size, MemTag mem_tag, const NativeCallStack& stack) {
+// Use pre-init if too early for NMT and check for size overflow and if we are still under MallocLimit
+size_t os::pre_init_and_check_size(void** raw_ptr, void* old_ptr, size_t size, MemTag mem_tag) {
+  // On malloc(0), implementations of malloc(3) have the choice to return either
+  // null or a unique non-null pointer. To unify libc behavior across our platforms
+  // we chose the latter.
+  size = MAX2((size_t)1, size);
 
   // Special handling for NMT preinit phase before arguments are parsed
-  void* rc = nullptr;
-  if (NMTPreInit::handle_malloc(&rc, size)) {
```

Comment:
So it looks like this was the only call to handle_malloc.  Is it now unused since malloc and realloc() call this function?

Reply:
Correct, we would no longer need to use this.